### PR TITLE
[RDY] Make computing price impact warnings local player specific

### PR DESCRIPTION
**Pull request by tobylane**
**Describe what the proposed change does**
- Only advise the local hospital of the price level impacts.

### CODE REVIEW

**Comment by lewri** [comment]
Is this something that should actually be moved into player_hospital? (maybe out of scope though).

**Comment by TheCycoONE** [comment]
Sort of. The price impact belongs where it is. There should be a stub method for advising for the hospital, and one that calls   self.world.ui.adviser:say in player_hospital. Then all these isPlayerHospital checks go away.

And yes that should replace this PR.

**Comment by tobylane** [comment]
Rewrote it that way.

**Comment by tobylane** [comment]
Ready.

File: CorsixTH/Lua/hospitals/player_hospital.lua [hunk 1]
```diff
@@ -398,6 +398,24 @@ function PlayerHospital:onEndMonth()
...
+--!param judgment (string - under, over, fair) The judgment of the price,
+-- from Hospital:computePriceLevelImpact
+--!param name (string) The name of the casebook entry of the diagnosis or disease
+function PlayerHospital:advisePriceLevelImpact(judgment, name)
+  local message
+  if judgment == "under" then
+    message = _A.warnings.low_prices:format(name)
+  elseif judgment == "over" then
+    message = _A.warnings.high_prices:format(name)
+  elseif judgment == "fair" then
```
Comment (Alberth289346) [inline CorsixTH/Lua/hospitals/player_hospital.lua:411]:
```suggestion
  else
    assert(judgment == "fair")
```
This ensures the code dies on typos in the `judgment` value.

Comment (TheCycoONE) [reply CorsixTH/Lua/hospitals/player_hospital.lua]: Stylistically I prefer
```lua
if foo == "a" then
   ---
elseif foo == "b" then
  ---
elseif foo == "c" then
  ---
else
  error("foo must be a, b, or c")
end
```
but I suppose it doesn't matter.

**Review by lewri — APPROVED** [review]
Could you resolve the conflict then it can be merged.

**Pull request merged**